ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -997,8 +1017,9 @@ public void renamePartition(Table tbl, Map<String, String> oldPartSpec, Partitio
       }
       String validWriteIds = null;
       if (AcidUtils.isTransactionalTable(tbl)) {
+
         TableSnapshot tableSnapshot;
-        if (replWriteId > 0) {
+        if (inReplication(tbl)) {
```

Comment:
Again, why is this change needed?

Reply:
Rebased with master and only kept relevant changes.